Snouter.Api/Snouter.Api/Controllers/IdentityController.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.IdentityModel.Tokens;
+using Snouter.Contracts.Requests;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text.Json;
+using System.Text;
+
+namespace Snouter.Api.Controllers
+{
+    [ApiController]
+    public class IdentityController : ControllerBase
+    {
+        private const string TokenSecret = "DumpDevInternship2023=<3"; // symmetric token secret
+        private static readonly TimeSpan TokenLifetime = TimeSpan.FromHours(8);
+
+        [HttpPost("token")]
+        public IActionResult GenerateToken(
+         [FromBody] TokenGenerationRequest request)
+        {
+            var tokenHandler = new JwtSecurityTokenHandler();
+            var key = Encoding.UTF8.GetBytes(TokenSecret);
+
+            var claims = new List<Claim>
+        {
+            new(JwtRegisteredClaimNames.Jti, Guid.NewGuid().ToString()),
+            new(JwtRegisteredClaimNames.Sub, request.Email),
+            new(JwtRegisteredClaimNames.Email, request.Email),
+            new("userid", request.UserId.ToString()),
+        };
+
+            foreach (var claimPair in request.CustomClaims)
+            {
+                var jsonElement = (JsonElement)claimPair.Value;
+                var valueType = jsonElement.ValueKind switch
+                {
+                    JsonValueKind.True => ClaimValueTypes.Boolean,
+                    JsonValueKind.False => ClaimValueTypes.Boolean,
+                    JsonValueKind.Number => ClaimValueTypes.Double,
+                    _ => ClaimValueTypes.String
+                };
+
+                var claim = new Claim(claimPair.Key, claimPair.Value.ToString()!, valueType);
+                claims.Add(claim);
+            }
+
+            var tokenDescriptor = new SecurityTokenDescriptor
+            {
+                Subject = new ClaimsIdentity(claims),
+                Expires = DateTime.UtcNow.Add(TokenLifetime),
+                Issuer = "https://id.internship-webapi.com",
+                Audience = "https://posts.internship-weapi.com",
```

Comment:
you should get these values (also token secret) from `appsettings.json` to make code a bit more reuseable and to make configuration easier to find and change

https://stackoverflow.com/a/31453663